tests/test_keystone.py

```diff
@@ -18,45 +20,49 @@
     'mips': b"addi $1, $1, 0xf"
 }
 
-def run_keystone(arch):
-    proj_arch = arch
-    is_thumb = False
-    if arch == "armel_thumb":
-        is_thumb = True
-        proj_arch = "armel"
-    p = angr.Project(os.path.join(test_location, proj_arch, "fauxware"), auto_load_libs=False)
-    addr = p.loader.main_object.get_symbol('authenticate').rebased_addr
 
-    sm = p.factory.simulation_manager()
-    if arch in ['i386', 'x86_64']:
-        sm.one_active.regs.eax = 3
-    else:
-        sm.one_active.regs.r1 = 3
+class TestKeystone(unittest.TestCase):
+    def run_keystone(self, arch):
+        proj_arch = arch
+        is_thumb = False
+        if arch == "armel_thumb":
+            is_thumb = True
+            proj_arch = "armel"
+        p = angr.Project(os.path.join(test_location, proj_arch, "fauxware"), auto_load_libs=False)
+        addr = p.loader.main_object.get_symbol('authenticate').rebased_addr
 
-    if is_thumb:
-        addr |= 1
-    block = p.factory.block(addr, insn_text=insn_texts[arch], thumb=is_thumb).vex
+        sm = p.factory.simulation_manager()
+        if arch in ['i386', 'x86_64']:
+            sm.one_active.regs.eax = 3
+        else:
+            sm.one_active.regs.r1 = 3
 
-    nose.tools.assert_equal(block.instructions, 1)
+        if is_thumb:
+            addr |= 1
+        block = p.factory.block(addr, insn_text=insn_texts[arch], thumb=is_thumb).vex
 
-    sm.step(force_addr=addr, insn_text=insn_texts[arch], thumb=is_thumb)
+        assert block.instructions == 1
 
-    if arch in ['i386', 'x86_64']:
-        nose.tools.assert_equal(sm.one_active.solver.eval(sm.one_active.regs.eax), 0x12)
-    else:
-        nose.tools.assert_equal(sm.one_active.solver.eval(sm.one_active.regs.r1), 0x12)
+        sm.step(force_addr=addr, insn_text=insn_texts[arch], thumb=is_thumb)
 
-def test_keystone():
+        if arch in ['i386', 'x86_64']:
+            assert sm.one_active.solver.eval(sm.one_active.regs.eax) == 0x12
+        else:
+            assert sm.one_active.solver.eval(sm.one_active.regs.r1) == 0x12
 
-    # Installing keystone on Windows is currently a pain. Fix the installation first (may it pip installable) before
-    # re-enabling this test on Windows.
-    if not sys.platform.startswith('linux'):
-        raise nose.SkipTest()
+    @skip_if_not_linux
+    def test_keystone(self):
 
-    for arch_name in insn_texts:
-        yield run_keystone, arch_name
+        # Installing keystone on Windows is currently a pain. Fix the installation first (may it pip installable) before
+        # re-enabling this test on Windows.
+        if not sys.platform.startswith('linux'):
+            return
+
+        for arch_name in insn_texts:
+            yield run_keystone, arch_name
```

Comment:
I believe this should be `self.run_keystone`, and that is what is causing the weird CI failure